### DISCUSSION TITLE
how to convert .mf4 to .mat

### DISCUSSION THREAD

**Discussion by FelixT80** [danielhrisca/asammdf#1055] [Q&A]
Hello everyone,

I tried to convert a mf4 file into Matlab .mat with the GUI, it worked well.
Than I tried to do this in script. But I'm not sure if I use the right code. I have read the documentation, but unfortunatelly that didn't helped me in detail.

My problem is, the following code runs without error but no mat-file is produced? What am I doing wrong? I hope you can help me.

I'm using the following code:

`import asammdf`
`from asammdf import MDF, Signal`
`import platform`
`import sys`
`from pprint import pprint`

`pprint("python=" + sys.version)`
`pprint("os=" + platform.platform())`

`try:`
`    import numpy`
`    pprint("numpy=" + numpy.__version__)`
`except ImportError:`
`    pass`

`try:`
`    import asammdf`
`    pprint("asammdf=" + asammdf.__version__)`
`except ImportError:`
`    pass`

`mdf = asammdf.MDF('C:/a/test.mf4')`
`mdf.export('mat', 'C:/a/test.mat')`

It produces following output:
C:\a\asammdf> & C:/Users/xyz/AppData/Local/Programs/Python/Python312/python.exe c:/a/conv.py
ldf is not supported
xls is not supported
xlsx is not supported
yaml is not supported
('python=3.12.3 (tags/v3.12.3:f6650f9, Apr  9 2024, 14:05:25) [MSC v.1938 64 '
 'bit (AMD64)]')
'os=Windows-10-10.0.19045-SP0'
'numpy=1.26.4' 
'asammdf=7.4.5'

but: no output file will has been created :-(

**Comment by JakubSchwenkbeck** [comment]
Hey Felix, I looked into your problem:
firstly
Check File Paths: Make sure the file paths are correct and that the script has write permissions to the target directory.

Check MDF File: Ensure that the 'C:/a/test.mf4' file is a valid MDF file and that it contains data. You can open it with the GUI again to confirm.

Verbose Output: Add more verbose output to get insights on what's happening during the export process.

Export Format Specification: Ensure the format string is correct. The export format should be specified correctly (in this case, 'mat').


This could be a better working code (didnt test tho) :
`import asammdf
from asammdf import MDF, Signal
import platform
import sys
from pprint import pprint
import os

pprint("python=" + sys.version)
pprint("os=" + platform.platform())

try:
    import numpy
    pprint("numpy=" + numpy.__version__)
except ImportError:
    pass

try:
    import asammdf
    pprint("asammdf=" + asammdf.__version__)
except ImportError:
    pass

input_file = 'C:/a/test.mf4'
output_file = 'C:/a/test.mat'

# Check if input file exists
if not os.path.exists(input_file):
    pprint(f"Input file does not exist: {input_file}")
else:
    pprint(f"Input file exists: {input_file}")

# Load the MDF file
try:
    mdf = asammdf.MDF(input_file)
    pprint("MDF file loaded successfully")
except Exception as e:
    pprint(f"Failed to load MDF file: {e}")

# Export to MAT file
try:
    mdf.export(fmt='mat', filename=output_file)
    pprint(f"Exported to MAT file successfully: {output_file}")
    # Check if output file was created
    if os.path.exists(output_file):
        pprint(f"Output file created: {output_file}")
    else:
        pprint("Output file was not created")
except Exception as e:
    pprint(f"Failed to export MAT file: {e}")
`This script adds several checks:

Verifies the existence of the input MDF file.
Attempts to load the MDF file and handles potential exceptions.
Attempts to export to a MAT file and handles potential exceptions.
Checks if the output file was successfully created.
By running this script, you should get more detailed information about where the process might be failing. If the input file exists and loads correctly, but the export still does not produce a file, the verbose output should help identify the issue.

I hope this helps in any way,
Have fun, Jakub

**Comment by FelixT80** [comment]
Hi Jakub,

Thank you for your support. Unfortunatelly I got the same result, no output file was created.
So to be sure that the input file mf4 is ok I created a example mf4 using this script:
https://asammdf.readthedocs.io/en/latest/examples.html
--> 1st example "Working with MDF".
[my_new_file.zip](https://github.com/user-attachments/files/16348237/my_new_file.zip)
But that doesnt helped.

Main question: Are the following lines correct to convert an mf4 to matlab? Is the syntax ok?
```
input_file = 'C:/a/my_new_file.mf4'
output_file = 'C:/a/my_new_file.mat'
mdf = asammdf.MDF(input_file)
mdf.export(fmt='mat', filename=output_file)
```

I got the following output from your script:
  ldf is not supported
  xls is not supported
  xlsx is not supported
  yaml is not supported
  ('python=3.12.3 (tags/v3.12.3:f6650f9, Apr  9 2024, 14:05:25) [MSC v.1938 64 '
   'bit (AMD64)]')
  'os=Windows-10-10.0.19045-SP0'
  'numpy=1.26.4'
  'asammdf=7.4.5'
  'Input file exists: C:/a/my_new_file.mf4'
  'MDF file loaded successfully'
  'Exported to MAT file successfully: C:/a/my_new_file.mat'
  'Output file was not created'

kind regards, Felix

**Comment by JakubSchwenkbeck** [comment]
Hey, so from what I see there is no issue with your syntax, the only problem I could see is the slashes and backslashes in the path. Im not quite sure how the asammdf lib wants you to input paths, but the convention would be to put backslashes, so rather C:\a\ . 
If it's not this, I really think there must be an issue with the export, as it doesn't create the file properly.
Let me now if you find the issue or atleast fix parts of it:)
Kind regards, Jakub

**Reply by FelixT80** [reply]
After debugging mdf.py i found the error, scipy was not installed on my machine.
After "pip install scipy" now it works :-)
